biofefi/components/synthetic_data.py

```diff
@@ -0,0 +1,150 @@
+                return pd.DataFrame(X), pd.Series(y, name="target")
+            except Exception as e:
+                raise ValueError(
+                    f"Error creating synthetic data for classification: {e}"
+                )
+
+    def _create_regression_data(self) -> None:
+        """
+        Creates synthetic data for regression problems.
+
+        Returns:
+            - Tuple: The synthetic data and labels.
+
+        Raises:
+            - ValueError: If an error occurs during the data creation.
+        """
+
+        if self._problem_type == ProblemTypes.Regression:
+            self._logger.info("Requested to create synthetic data for Regression...")
+
+            try:
+                X, y = make_regression(
+                    n_samples=self.synthetic_options.n_samples,
+                    n_features=self.synthetic_options.n_features,
+                    shuffle=self.synthetic_options.is_shuffle,
+                    random_state=self.synthetic_options.random_state,
+                    n_informative=self.synthetic_options.reg_n_informative,
+                    n_targets=self.synthetic_options.reg_n_targets,
+                    bias=self.synthetic_options.reg_bias,
+                    noise=self.synthetic_options.reg_noise,
+                    coef=self.synthetic_options.reg_coef,
+                )
+                return pd.DataFrame(X), pd.Series(y, name="target")
+            except Exception as e:
+                raise ValueError(f"Error creating synthetic data for regression: {e}")
+
+    def data_initialisation(self) -> None:
+        """
+        Initialises the synthetic data creation.
+
+        Raises:
+            - InitialisationError: If an error occurs during
+            the initialisation.
+        """
+        self.initialise = True
+        self._logger.info("Initialising synthetic data creation...")
+
+        try:
+            # create synthetic data for classification or regression
+            if self._problem_type == ProblemTypes.Classification:
+                self.X, self.y = self._create_classification_data()
+            elif self._problem_type == ProblemTypes.Regression:
+                self.X, self.y = self._create_regression_data()
+            else:
+                raise ValueError(f"Unsupported problem type: {self._problem_type}")
+
+            # Combine features and labels
+            synthetic_data = pd.concat([self.X, self.y], axis=1)
+
+            # Perform assertions on the generated data
+            data_checker = DataLoaderChecker(self.X, self.y)
+            data_checker.perform_data_checks()
+
+            # Save the synthetic data to a csv file
+            synthetic_data.to_csv(self._output_file_name, index=False)
+            self._logger.info(f"Synthetic data saved to {self.output_file}")
+
+        except Exception as e:
+            raise InitialisationError(f"An Error initialising synthetic data: {e}")
+
+
+class InitialisationError(Exception):
```

Comment:
```suggestion
class SyntheticDataInitialisationError(Exception):
```
`InitialisationError` sounds too generic

Reply:
Ok Dan. I forget in the last commit I guess. My bad